### PR TITLE
comment

### PR DESCRIPTION
**Pull request by candelarialopezz**

### CODE REVIEW

File: app/assets/stylesheets/application.sass.scss [hunk 1]
```diff
@@ -9,3 +10,5 @@
 @import 'devise/edit';
 @import 'user/index';
 @import 'user/follow';
+// @import 'font-awesome-sprockets';
+// @import 'font-awesome';
```
Comment (EnzoCozza) [inline app/assets/stylesheets/application.sass.scss:14]: If you are not using these, remove them

File: app/assets/stylesheets/post/_show.scss [hunk 2]
```diff
@@ -26,3 +26,96 @@
...
+    display: none;
+}
+
+.new-comment-row {
+    margin-left: 45px;
+    margin-right: 30px;
+    padding-right: 30px;
+    align-items: center;
+    max-width: 515px;
+    //border-bottom: solid 1px;
```
Comment (EnzoCozza) [inline app/assets/stylesheets/post/_show.scss:80]: Remove this

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:content)
+  end
+
+  def set_commentable(commentable_type, commentable_id)
+    @commentable = commentable_type.constantize.find(commentable_id)
+    @comment.commentable = @commentable
```
Comment (EnzoCozza) [inline app/controllers/comments_controller.rb:24]:
```suggestion
    @comment.commentable = commentable_type.constantize.find(commentable_id)
```

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  def create
+    @comment = Comment.new(
+      user: current_user,
+      content: params[:comment][:content]
```
Comment (EnzoCozza) [inline app/controllers/comments_controller.rb:7]: Let's use `comment_params.merge(user: current_user)`

File: app/javascript/comment.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    if (event.which === 13) { // 13 is the key code for Enter
+      addComment();
+    }
+    function addComment() {
+      const commentText = $('#post_comment').val();
+      const parentId  = $('#post_comment').data('parent-id');
+
+      if (commentText === '') {
+        return;
+      }
```
Comment (EnzoCozza) [inline app/javascript/comment.js:15]: Let's show an error message to the user that content cannot be empty

File: app/javascript/comment.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            const commentRowAbove = commentsComments.previousElementSibling;
+            const linkelement = commentRowAbove.querySelector('.link-to-comments');
+            const commentsCount = linkelement.querySelector('.comment-count');
+            let currentCount = parseInt(commentsCount.textContent, 10) || 0;
+            currentCount += 1;
+            commentsCount.textContent = currentCount;
+  
+            commentsComments.style.display = 'block';
+  
+            $(event.target).val('');
```
Comment (EnzoCozza) [inline app/javascript/comment.js:85]: There's double indentation in this block

File: app/javascript/comment.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            const tempContainer = document.createElement('div');
+            tempContainer.innerHTML = data.attachment_partial;
+            commentsComments.insertBefore(tempContainer.firstChild, commentsComments.firstChild);
+            
+  
+            const commentRowAbove = commentsComments.previousElementSibling;
+            const linkelement = commentRowAbove.querySelector('.link-to-comments');
+            const commentsCount = linkelement.querySelector('.comment-count');
+            let currentCount = parseInt(commentsCount.textContent, 10) || 0;
+            currentCount += 1;
```
Comment (EnzoCozza) [inline app/javascript/comment.js:80]:
```suggestion
            currentCount++;
```

File: app/views/posts/_new_comment_row.html.haml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+.new-comment-row
+  .col-md-1.user-img-new-comment
+    = image_tag current_user.avatar, class: 'rounded-circle user-img-post m-0'
+  .col-md-11
+    %input.form-control.comment-input.comment_comment{type: "text", name: "comment", placeholder: "Write a comment...", data: { parent_id: parent_id } }
```
Comment (EnzoCozza) [inline app/views/posts/_new_comment_row.html.haml:5]:
```suggestion
    %input.form-control.comment-input.comment_comment{ type: "text", name: "comment", placeholder: "Write a comment...", data: { parent_id: parent_id } }
```

File: app/views/posts/show.html.haml [hunk 9]
```diff
@@ -1,5 +1,14 @@
...
   = render partial: 'post_content', locals: { post: @post, post_links: false }
+
+  .comments-container.d-flex.flex-column.align-items-center
+    = render partial: 'comment', collection: @post.comments.order(created_at: :desc)
+
+  .new-comment-row-post
+    .col-md-1
+      = image_tag current_user.avatar, class: 'rounded-circle user-img-post m-0'
+    .col-md-11
+      %input.form-control.comment-input#post_comment{type: "text", name: "comment", placeholder: "Write a comment...", data: { parent_id: @post.id } }
```
Comment (EnzoCozza) [inline app/views/posts/show.html.haml:14]:
```suggestion
      %input.form-control.comment-input#post_comment{ type: "text", name: "comment", placeholder: "Write a comment...", data: { parent_id: @post.id } }
```

File: spec/requests/comments_spec.rb
Comment (EnzoCozza) [inline spec/requests/comments_spec.rb:1]: Let's add a test to verify the newest comment has `comment.content` set as content, and a test to verify that the user must be logged in to leave a comment

File: spec/requests/comments_spec.rb [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+      end
+
+      it 'verifies posts comment count increases' do
+        count = post1.comments.count
+        post post_comments_path(post1),
+             params: { comment: { content: comment.content }, commentable_type: 'Post', commentable_id: post1.id }
+        expect(post1.comments.count).to be(count + 1)
+      end
+
+      it 'verifies las comment content is correct' do
```
Comment (daizuoheng) [inline spec/requests/comments_spec.rb:47]:
```suggestion
      it 'verifies last comment content is correct' do
```

File: spec/requests/comments_spec.rb [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+      end
+
+      it 'verifies posts comment count increases' do
+        count = comment.comments.count
+        post comment_comments_path(comment),
+             params: { comment: { content: comment.content }, commentable_type: 'Comment', commentable_id: comment.id }
+        expect(comment.comments.count).to be(count + 1)
+      end
+
+      it 'verifies las comment content is correct' do
```
Comment (daizuoheng) [inline spec/requests/comments_spec.rb:77]: Same here

**Review by daizuoheng — APPROVED** [review]

**Pull request merged**